utils/validate.ts

```diff
@@ -0,0 +1,28 @@
+interface ErrorInterface {
+  valid: boolean;
+  error: null | string;
+}
+
+export const invalidCharError: ErrorInterface = {
+  valid: false,
+  error: 'contains invalid characters'
+};
+export const validationSuccess = { valid: true, error: null };
+export const usernameTooShort = { valid: false, error: 'is too short' };
+export const usernameIsHttpStatusCode = {
+  valid: false,
+  error: 'is a reserved error code'
+};
```

Comment:
I would remove this `ErrorInterface` declaration. There is not much point in having it if:
a) It is not exported
b) Is only used to annotate a single object literal

I am surprised we do not have more strict type requirements for the return of functions.

Reply:
Done!

Reply:
It's been a while since I originally did this, but now that the build is failing, I'm reminded why I created this interface in the first place. The object returned has a property `error`, which can sometimes be null and other times string. Without the (now removed) interface, TypeScript isn't happy. I'm open to suggestions, but I don't see a way to fix it without including the interface.

Build failure:
https://github.com/freeCodeCamp/freeCodeCamp/actions/runs/3857032919/jobs/6574207466

Reply:
This is an issue with the way the test is written, not with the way the function is written. That is to say, the test code needs to be changed. Easiest being to just type the `expected` variable with `{ valid: boolean, error: null | string }`

Personally, I just refactoring the test to `expect` on each conditional branch - no global `expected` variable. This would be preferred because it would make use of the fact that a failure in any `expect` would actually point out the line/condition, as opposed to just saying "well, one of these conditions failed".

Hope this helps

Reply:
Thanks for the suggestion. Went with the easier approach :)